src/main/java/transaction/TransactionList.java

```diff
@@ -27,11 +30,30 @@ public static void addTx(Transaction transaction) {
             throw CarException.licensePlateNumberNotFound();
         }
 
+        for (Transaction tx : transactionList) {
+            if (tx.getCarLicensePlate().equals(licensePlateNumber) &&
+                    datesOverlap(tx.getStartDate(), tx.getEndDate(),
+                            transaction.getStartDate(), transaction.getEndDate())) {
+                System.out.println("Car " + licensePlateNumber +
+                        " is already rented during this period. Transaction not added.");
+                return;
+            }
+
+            if (tx.getCustomer().equalsIgnoreCase(customerName) &&
+                    datesOverlap(tx.getStartDate(), tx.getEndDate(),
+                            transaction.getStartDate(), transaction.getEndDate())) {
+                System.out.println("Customer " + customerName +
+                        " already has a rental during this period. Transaction not added.");
+                return;
```

Comment:
might wanna extract the if condition to be more clear

Reply:
Yes! Done